### ISSUE TITLE
Mul op returns wrong output when multiplied with NaN

### ISSUE THREAD

**Issue by ruthreshx** [tenstorrent/tt-metal#12776]
Mul op returns wrong output:

**Actual Output:**
NaN * 0.5 = returns 255211775190703847597530955573826158592.00000 

**Expected Output:**
NaN * 0.5 = NaN

Also we have tried the **nan_gs** value from the device 6.9752e19 * 0.5 it returns 34875875514357121024.00000

Note: I noticed one more thing, If NaN multiplies with any decimal value from 0.0 to 0.66 it returns above bigger value, otherwise it returns NaN as expected.

Observed in Both GS and WH

This issue blocks #6721 and #8634

**Comment by SeanNijjar** [comment]
Redirecting as I am not the right person to look at this

**Comment by ttmtrajkovic** [comment]
@ruthreshx 
Which op is this, please specify the test you were using and how to reproduce the problem

**Comment by ruthreshx** [comment]
Hi @ttmtrajkovic ,
Please find the below test to reproduce the issue.
Device: GS, WH300
Op: **MUL**

```
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

from loguru import logger
import random
import pytest
import torch
import ttnn

from tests.ttnn.utils_for_testing import assert_with_pcc
from tests.ttnn.python_api_testing.sweep_tests import ttnn_ops


def run_eltwise_mul_tests(input_shape, dtype, dlayout, in_mem_config, output_mem_config, data_seed, device):
    torch.manual_seed(data_seed)

    x = torch.Tensor(size=input_shape[0]).uniform_(0, 0.66).to(torch.bfloat16)
    y = torch.full(size=input_shape[1], fill_value=float('nan'))

    try:
        # get ref result
        ref_value = torch.mul(x, y)

        x = ttnn_ops.setup_ttnn_tensor(x, device, dlayout[0], in_mem_config, dtype[0])
        y = ttnn_ops.setup_ttnn_tensor(y, device, dlayout[1], in_mem_config, dtype[1])

        tt_result = ttnn.mul(x, y)
        tt_result = ttnn_ops.ttnn_tensor_to_torch(tt_result, output_mem_config)
        
        print("tt_result ==> ", tt_result)
        logger.info(f"Mul run {input_shape[0]} * {input_shape[1]} finished")

    except Exception as e:
        logger.warning(f"Operation execution crashed")
        raise e

    assert len(tt_result.shape) == len(ref_value.shape)
    assert tt_result.shape == ref_value.shape
    assert_with_pcc(ref_value, tt_result, 0.99)


test_sweep_args = [
    (
        [(32, 32), (32, 32)],
        [ttnn.bfloat16, ttnn.bfloat16],
        [ttnn.TILE_LAYOUT, ttnn.TILE_LAYOUT],
        (ttnn.DRAM_MEMORY_CONFIG),
        (ttnn.DRAM_MEMORY_CONFIG),
        17799073,
    ),
]


def test_eltwise_mul(device):
    for i in range(1):
        for input_shape, dtype, dlayout, in_mem_config, output_mem_config, data_seed in test_sweep_args:
            run_eltwise_mul_tests(input_shape, dtype, dlayout, in_mem_config, output_mem_config, data_seed, device)
```

**Comment by prajaramanTT** [comment]
@ttmtrajkovic Is this still an open issue ? If not, can you please close this ticket ? Thanks

**Comment by nvelickovicTT** [comment]
@ruthreshx just FYI, @atatuzunerTT will take a closer look into the issue reported here.

But, just to clarify one thing first. Looking at your example repro, you are setting one of the input tensors to NaN values.
If you look at Debugging section in Handling special numbers tech report (https://github.com/tenstorrent/tt-metal/blob/main/tech_reports/Handling_Special_Value/special_values.md), it says such cases are likely to not be detected in current architectures, and would need to be filtered by the host.

**Comment by atatuzunerTT** [comment]
I investigated this issue and also confirmed with the hardware team that the intended behavior follows the tech report above. The FPU doesn't have special handling for NaN values in its input operands. Therefore, any NaN value will be interpreted as an ordinary number. This explains the numbers in the test case.

The convention here is that the user is expected to filter out NaN values before sending the data into Tensix. However, there is NaN/Inf detection at the output of FPU/SFPU, recorded using status bits. This means that any NaN/Inf value produced during an intermediate operation will be detected and the user will be informed, so that they can take appropriate actions.

More detailed information can be found here: https://github.com/tenstorrent/tt-metal/issues/8944#issuecomment-2146237816

In short, this isn't a bug, but expected behavior. We can close this issue if you have no further questions.

**Comment by nvelickovicTT** [comment]
Closing as it is resolved. @ruthreshx , @cmaryanTT .
